src/Api.Rest/Common/Utilities/OAuthHelper.cs

```diff
@@ -74,6 +74,19 @@ public static string IdentityClaimsToFriendlyText( IList<Claim> claims )
 			return $"{name} ({email})";
 		}
 
+		/// <summary>
+		/// Extract the value of the JWT 'exp' token claim and convert it to the expiration date.
+		/// </summary>
+		/// <param name="claims">A collection of claims.</param>
+		/// <returns>The expiration date of the token.</returns>
+		public static DateTime ClaimsToExpirationTime( IList<Claim> claims )
+		{
+			var expClaim = claims.SingleOrDefault( claim => claim.Type == "exp" )?.Value;
+			return long.TryParse( expClaim, out var secondsSinceEpoch )
+				? DateTimeOffset.FromUnixTimeSeconds( secondsSinceEpoch ).DateTime

```

Comment:
Maybe `UtcDateTime` instead of `DateTime`? I think the value should be the same (offset is zero in both cases) but then the `DateTimeKind` is `Utc` instead of `Unspecified`. (I mean it should be equivalent to `expiration` in method `ChooseAccessToken`...?)

Reply:
This difference was indeed a little bit hidden, and it totally makes sense to also use UtcDateTime here.